tests/test_hwi_rpc.py

```diff
@@ -1,27 +1,77 @@
+""" This is just a manual test to understand how HWI works. All tests are marked skipped as hardware plugged in is necessary.
+    Don't take this as best practise. I doubt, this is just something to test difference in behaviour for migration from HWI 2.0.2 to 2.1.0
+"""
+
 import logging
+import io
+import pytest
 from cryptoadvance.specter.hwi_rpc import HWIBridge
 from cryptoadvance.specter.util.descriptor import Descriptor
 
 
-def test_enumerate(caplog):
+# @pytest.mark.skip()
+def test_trezor(caplog, monkeypatch):
+    """In order to get this test working, you have to run it with "-s":
+    pytest tests/test_hwi_rpc.py::test_enumerate_trezor  -vv -s
+    """
     caplog.set_level(logging.DEBUG)
 
     hwi = HWIBridge()
     # bla = hwi.detect_device()
 
     res = hwi.enumerate(passphrase="")[0]
-
+    print(res)
     # seems to be normal
     assert res["type"] == "trezor"
     assert res["model"] == "trezor_1"
-    assert res["path"] == "webusb:003:1:1:1:2"
-    assert res["error"].startswith(
-        "Could not open client or get fingerprint information: Passphrase needs to be specified before the fingerprint information can be retrieved"
-    )
+    assert res["path"].startswith("webusb:003:1:1:")
+
+    if res["needs_pin_sent"]:
+        assert res["error"].startswith(
+            "Could not open client or get fingerprint information: Trezor is locked"
+        )
+
+    else:
+        assert res["error"].startswith(
+            "Could not open client or get fingerprint information: Passphrase needs to be specified before the fingerprint information can be retrieved"
+        )
+        assert len(res["fingerprint"]) == 8
     assert res["code"] == -12
-    assert res["fingerprint"] == "1ef4e492"
 
+    res = hwi.prompt_pin(device_type="trezor", passphrase="")
+    assert res["success"] == True
+    # monkeypatch.setattr('sys.stdin', io.StringIO('my input'))
+    pin = input("Enter pin: ")
+
+    res = hwi.send_pin(pin, device_type="trezor", passphrase="")
+    assert res["success"] == True
     results = hwi.extract_xpubs(chain="test", device_type="trezor").split("\n")
     assert len(results) == 9
     assert results[0].startswith("[")
     assert results == None
+
+
+def test_jade(caplog):
+    caplog.set_level(logging.DEBUG)
+
+    hwi = HWIBridge()
+    # bla = hwi.detect_device()
+
+    res = hwi.enumerate(passphrase="")[0]
+
+    # seems to be normal
+    assert res["type"] == "jade"
+    assert res["model"] == "jade"
+    assert res["path"] == "/dev/ttyUSB0"
```

Comment:
when I use pyserial or hwi or blockstream green, my newer jade always shows this path:
```
> ./hwi enumerate
[
  {
    "type": "jade",
    "model": "jade",
    "path": "/dev/cu.usbserial-0224A7E3",
    "needs_pin_sent": false,
    "needs_passphrase_sent": false,
    "fingerprint": "..."
  }
]
```

Reply:
```
[
  {
    "type": "jade",
    "model": "jade",
    "path": "/dev/ttyUSB0",
    "needs_pin_sent": false,
    "needs_passphrase_sent": false,
    "fingerprint": "...."
  }
]
```
This is mine, an older one. I don't think that the path is specific to the version. But i think these two numbers are:
```
$ ls -l /dev/ttyUSB0                 
crw-rw----+ 1 root plugdev 188, 0 Mai 18 14:35 /dev/ttyUSB0
$
```
(188 / 0).